enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -511,13 +540,29 @@ class AdjointGenerator
           ts = setPtrDiffe(orig_ptr, Constant::getNullValue(valType), Builder2);
           addToDiffe(orig_val, dif1, Builder2, FT);
         }
+        break;
+      }
+      case DerivativeMode::ForwardMode: {
+        IRBuilder<> Builder2(&SI);
+        getForwardBuilder(Builder2);
+
+        if (constantval) {
+          ts = setPtrDiffe(orig_ptr, Constant::getNullValue(valType), Builder2);
```

Comment:
I don't think this set if load is inactive is correct. I think no store should happen in that case.

E.g. consider

```
%unused = load %x
%toret = load %x
ret %toret
```

The unused value would cause d_x to be set to zero, making the eventual derivative zero, which seems incorrect.

Reply:
This is a store

Reply:
Ah my mistake, this is fine then.